homeassistant/components/light/xiaomi_miio.py

```diff
@@ -728,3 +744,83 @@ def __init__(self, name, light, model, unique_id):
         except DeviceException as ex:
             self._available = False
             _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+
+class XiaomiPhilipsMoonlightLamp(XiaomiPhilipsBulb):
+    """Representation of a Xiaomi Philips Zhirui Bedside Lamp."""
+
+    def __init__(self, name, light, model, unique_id):
+        """Initialize the light device."""
+        super().__init__(name, light, model, unique_id)
+
+        self._hs_color = None
+        self._state_attrs.pop(ATTR_DELAYED_TURN_OFF)
+        self._state_attrs.update({
+            ATTR_SLEEP_ASSISTANT: None,
+            ATTR_SLEEP_OFF_TIME: None,
+            ATTR_TOTAL_ASSISTANT_SLEEP_TIME: None,
+            ATTR_BRAND_SLEEP: None,
+            ATTR_BRAND: None,
+        })
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        return 153
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        return 588
+
+    @property
+    def hs_color(self) -> tuple:
+        """Return the hs color value."""
+        return self._hs_color
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP
+
+    async def async_update(self):
+        """Fetch state from the device."""
+        from miio import DeviceException
+        try:
+            state = await self.hass.async_add_executor_job(self._light.status)
+            _LOGGER.debug("Got new state: %s", state)
+
+            self._available = True
+            self._state = state.is_on
+            self._brightness = ceil((255 / 100.0) * state.brightness)
+            self._color_temp = self.translate(
+                state.color_temperature,
+                CCT_MIN, CCT_MAX,
+                self.max_mireds, self.min_mireds)
+            self._hs_color = self._get_hs(state.rgb)
+
+            self._state_attrs.update({
+                ATTR_SCENE: state.scene,
+                ATTR_SLEEP_ASSISTANT: state.sleep_assistant,
+                ATTR_SLEEP_OFF_TIME: state.sleep_off_time,
+                ATTR_TOTAL_ASSISTANT_SLEEP_TIME:
+                    state.total_assistant_sleep_time,
+                ATTR_BRAND_SLEEP: state.brand_sleep,
+                ATTR_BRAND: state.brand,
+            })
+
+        except DeviceException as ex:
```

Comment:
We should only wrap the call that could raise with the `try... except`. Move this up and return in the end of the `except` block.

Reply:
Fixed.